chain/jsonrpc/src/api/transactions.rs

```diff
@@ -52,12 +57,57 @@ impl RpcFrom<TxStatusError> for RpcTransactionError {
     }
 }
 
-fn decode_signed_transaction(value: Value) -> Result<SignedTransaction, RpcParseError> {
-    #[serde_as]
-    #[derive(serde::Deserialize)]
-    struct Payload(#[serde_as(as = "(Base64,)")] (Vec<u8>,));
-
-    let Payload((bytes,)) = Params::<Payload>::parse(value)?;
+fn decode_signed_transaction(value: String) -> Result<SignedTransaction, RpcParseError> {
+    let bytes = from_base64(&value)
+        .map_err(|err| RpcParseError(format!("Failed to decode transaction: {}", err)))?;
     SignedTransaction::try_from_slice(&bytes)
         .map_err(|err| RpcParseError(format!("Failed to decode transaction: {}", err)))
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::api::RpcRequest;
+    use near_jsonrpc_primitives::types::transactions::{
+        RpcBroadcastTransactionRequest, RpcTransactionStatusCommonRequest,
+    };
+    use near_primitives::borsh::BorshSerialize;
+    use near_primitives::hash::CryptoHash;
+    use near_primitives::serialize::to_base64;
+    use near_primitives::transaction::SignedTransaction;
+    use std::str::FromStr;
+
+    #[test]
+    fn test_serialize_tx_status_params_as_vec() {
+        let params =
+            serde_json::json!(["6zgh2u9DqHHiXzdy9ouTP7oGky2T4nugqzqt9wJZwNFm", "sender.testnet"]);
+        assert!(RpcTransactionStatusCommonRequest::parse(params).is_ok());
+    }
+
+    #[test]
+    fn test_serialize_tx_status_params_as_object() {
+        let params = serde_json::json!({"hash": "6zgh2u9DqHHiXzdy9ouTP7oGky2T4nugqzqt9wJZwNFm", "account_id": "sender.testnet"});
```

Comment:
We may rename them btw, they may be "tx_hash" and "sender_account_id".
I just decided not to add new naming options to our collection.